webapp/src/hooks/useWeb3Wallets.ts

```diff
@@ -0,0 +1,176 @@
+import { providers } from "ethers";
+import {
+  EthereumWallet,
+  isEthereumWallet,
+  isSolanaWallet,
+  SolanaWallet,
+  Wallet,
+} from "../models/wallets";
+import {
+  Wallet as SolanaWalletInterface,
+  WalletName,
+} from "@solana/wallet-adapter-wallets";
+import { Chains, useChain } from "./chainContext";
+import { useCallback, useEffect, useState } from "react";
+import { useWeb3React as useEVMWallet } from "@web3-react/core";
+import { useWallet as useSolanaWallet } from "@solana/wallet-adapter-react";
+import { AbstractConnector } from "@web3-react/abstract-connector";
+import {
+  getWalletConnectConnector,
+  injectedConnector,
+  walletlinkConnector,
+} from "shared/lib/utils/connectors";
+import { CHAINS_TO_ID } from "../constants/constants";
+import { switchChains } from "shared/lib/utils/chainSwitching";
+
+interface Data {
+  chainId: number;
+  account: string;
+  isActive: boolean;
+
+  activate: () => Promise<void>;
+  deactivate: () => Promise<void>;
+}
+
+interface EVMWalletData extends Data {
+  provider: providers.Web3Provider;
+}
+
+interface SolanaWalletData extends Data {
+  wallet: SolanaWalletInterface;
+}
+
+type WalletTypes = EVMWalletData | SolanaWalletData;
+
+interface Web3Data {
+  wallets: Array<WalletTypes>;
+  addNewChain: (wallet: Wallet, chain: Chains) => Promise<void>;
+}
+
+const evmConnectors: Record<EthereumWallet, () => AbstractConnector> = {
+  [EthereumWallet.Metamask]: () => injectedConnector,
+  [EthereumWallet.WalletConnect]: getWalletConnectConnector,
+  [EthereumWallet.WalletLink]: () => walletlinkConnector,
+};
+
+export const useWeb3Data = (): Web3Data => {
```

Comment:
@kenchangh , this is the refactored `useWeb3Data` hook that has all the changes that are needed to integrate into the existing `useWeb3Wallet`. The only change made here is related to how a chain is "activated" when selected in the network switcher. 

All other references still point back to the original hook, only the `addNewChain` function found in this file is used to replace the existing `activate` function.